app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,45 @@
+class GroupsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @categories = Group.all
```

Comment:
- [ ] As per the requirement, the data should be private for each user. Please fix this. Right now, you are rendering all data and there is no privacy.
![image](https://github.com/ahmed025566/cost_commander/assets/91486791/2c533f13-43c8-4fa5-8571-e1933138e24c)